### ISSUE TITLE
Use Jekyll

### ISSUE THREAD

**Issue by piyush-kurur** [cseAtIITK/pg-website#24]
Are we planning to move to jekyll (or some such static website). It would be good if we do
that. We can also host a version of the page on github for preview.

**Comment by adarshaj** [comment]
Yes sir, but this would take a longer time. I'll start work on this in a different branch and get some more help from others to accomplish. 
For hosting, its straight forward as of now, i'll create a new gh-pages branch and it'll be immediately available. We have to make changes to it once jekyll integartion is complete.

**Comment by adarshaj** [comment]
@piyush-kurur - I've configured the repo to host pages via `master` branch - It can be viewed at 
https://cseatiitk.github.io/pg-website/

**Comment by piyush-kurur** [comment]
Also I was wondering whether it would be a good idea to work on the cse home page as well. The pg admissions can then be just a subsite of the main page.

On the other hand if one wants to keep the admission related stuff separate (previledge separation for example). Then the following design would be good


main page - contains all the styling stuff and templates etc
each activity like admissions for example has a separate repository which contains only the contents.
the main repository includes subsites like admissions as git submodules.

**Comment by adarshaj** [comment]
I think its a very solid idea sir! But this needs to be taken up by webmaster (prof. Gaurav?)? I think if we move all the repos here, there should be some way of continuous integration running which deploys the changes as soon as its merged to master (may be with some sanity checks).

**Comment by piyush-kurur** [comment]
CI  + automatic deployment might work well now that we have a jenkins server in the dept. I am not sure
but is the jenkins server available from outside the IITK network @sakshamsharma can confirm.

**Comment by adarshaj** [comment]
I think its not much of a concern as Jenkins can also do polling from within iitk (we just need proxy / internet to be accessible on jenkins). But I do not know the details about the CI and where its hosted etc.

**Comment by sakshamsharma** [comment]
We don't necessarily need an internal CI for this job, although it can be easily done. 

The jenkins is not available outside IITK (it is, but we've blocked access from non-iitk IPs).  It can be changed if needed.

**Comment by adarshaj** [comment]
@sakshamsharma - we do need internal CI because the website has to be deployed on internal server at turing.
